### ISSUE TITLE
Encourage users to write math in a more accessible way

### ISSUE THREAD

**Issue by mkorje** [typst/typst#8263]
### Description

We should encourage best practices in the math docs for writing math that is accessible. For example, the use of `intent` and `arg` attributes (once #8262 is complete), and the use of invisible operators (e.g. the "apply function" operator (`&#x2061;`) should be used to indicate function application).

See [MathML 4, C. MathML Accessibility](https://www.w3.org/TR/mathml4/#accessibility) for more.

**Comment by mkorje** [comment]
See https://github.com/typst/codex/issues/45 for progress on adding names for the invisible operators.

**Comment by reknih** [comment]
I think we need to be clear here what the scope for this is. I think that there is a meaningful tradeoff between ease of authoring and accessibility.

For example, in the document [MathML Accessibility Gap Analysis](https://w3c.github.io/mathml-docs/gap-analysis/), W3C Math WG editors suggest this:

> There is a supposition that semantic readings are “better”, but this has not been confirmed by research for people that are blind; studies do show that semantic reading styles are better for individuals with dyslexia and other non-visual print disabilities.

I think that this gives us a clear framework along which we can design Typst and give guidance:

- **Level 1:** Avoid serializations in which off-the-shelf AT makes a wrong semantic assumption when announcing math. For example, when an `mtable` element is used to represent an equation system, AT may instead mistakenly assume that that `mtable` is really a matrix and mistakenly announce the equation system as such.
- **Level 2:** Avoid serializations where AT would literally read the notation when announcing the expression using other, semantic terms would be customary. This includes announcing `(x, y)` as _the point x comma y_ instead of _open paren x comma y close paren._

Since math in Typst, MathML Core, and presentational MathML 3 are tools to express math notation, not mathematical expressions, interpreting their rendered output normally assumes the sighted reader is familiar enough with the expression's subject's conventions in order to interpret the notation semantically. Likewise, a sighted reader not familiar with a subject matter cannot intuit the correct way to pronounce $\binom{n}{m}$ in the field of stochastic math; they would likely assume this is a vector and not announce the binomial coefficient as _n over m._

In that sense, I'd like to argue that what AT announces is more like what you see than what you would read the equation out as a practitioner -- interpretation may still be required. A failure to meet level 1 would result in presentational differences that actually result in different semantics not correctly conveyed to the AT user, which, I'd say, renders the document inaccessible. 

From a position of extensibility, too, we should make announcing notation without semantics a good experience. If a paper invents a new formalism alongside notation, it will define to the reader how to interpret it, even if existing AT mechanisms may assume a closed set of intents or semantics in which the new formalism is not expressible. Of course, it would be best to allow the paper authors to define new semantics and output modes for their new notations, but existing AT may or may not support this.

Level 2, conversely, contains additional information not necessarily obvious from the graphical representation. If we assume that all readers of the document are familiar with the field and its notation, having AT announce math as practitioners would is a nice affordance to readers with e.g. dyslexia, as the W3C WG states above. However, when the document has an educational intent, correct and semantic announcements have a pedagogical dimension. They add information not immediately obvious to sighted readers that may need to be stated somewhere else for sighted, e.g. in prose. Alternatively, the document could be regarded as a mere part of a course with in-person instruction that explains the notation. Documents may also want to expose their AT users to literal readings of the notation, so that the reader can consume non-marked up sources surely distributed elsewhere. At any rate, educational documents need to make sure to not fail [WCAG 2.1 Success Criterion 1.3.1 Info and Relationships](https://www.w3.org/WAI/WCAG21/Understanding/info-and-relationships.html) and make different information available to AT and non-AT users.

Hence, I think that our aim should be to provide syntax and documentation to make reaching level 1 easy and the norm while providing tools for authors to reach level 2 where appropriate. IMO, level 1 is the right tradeoff between accessibility and ease of authoring for most cases.

Finally, I think that our tooling should take care to avoid degrading AT output by annotating partial subtrees: If a particular element has a defined intent but its children default to `:legacy` behavior, the overall AT output for the expression may be worse than if no intents were present at all. Any guidance we give should be tested against NVDA, VoiceOver, and Orca.
